### DISCUSSION TITLE
Error: src refspec main does not match

### DISCUSSION THREAD

**Discussion by mcnaveen** [community/community#22766] [New to GitHub]
<p>Can’t push to the main branch. Default Branch is main now.</p>
<p><img src="https://user-images.githubusercontent.com/8493007/181073073-c81d0d87-d7d9-4096-8839-3af5b3206d93.png" alt="image" data-base62-sha1="lctPeiKM5CLfglrepWRuinYtUly" width="1" height="1"></p>
<p>Image: <a href="https://imgur.com/a/xlf2Pcz" rel="noopener nofollow ugc">https://imgur.com/a/xlf2Pcz</a></p>
<pre><code class="lang-auto">ninja@ninja:~/Scripts/nodejs-api$ git push origin main
error: src refspec main does not match any
error: failed to push some refs to 'https://github.com/mcnaveen/API'
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<p>Do you have a local <code>main</code> branch, too? The error message sounds like you don’t.</p>

**Comment by mcnaveen** [comment]
<p>Solved by running <code>git checkout -b main</code> then <code>git push origin main</code></p>
<p>You saved me <a class="mention" href="/airtower-luna">@airtower-luna</a> Thanks.</p>

**Comment by billionaire87** [comment]
This error simply means that you don't have branch in current repo called main in your local git. but in the remote git it exist, because GitHub will create a default branch **_main_** in the repo for you . So in other to solve the error you need to create a new branch called man in your local git before you can push it to the specified origin. note that the default branch name in the local repo is "master" hence you can use master without any problems because It simply exits. _but does it exist in the remote_  **NO ** but github will create a new branch on the remote repo called master for you. So in summary

1. Create a new branch  _git checkout -b main_

2. Do this to prevent errors  _git pull --rebase origin main_

3. Lastly push  _git push -u origin main_

Dont forget that this will use the default Branch called main on your remote repo but if you go with the branch named master, it will create a new branch called _master_ on your remote repo , and leave the main default branch blank

Thank you.

**Reply by mcnaveen** [reply]
Thanks @billionaire87! :tada:
